dynamic_instrumentation/src/instrument.rs

```diff
@@ -281,49 +278,30 @@ impl<'tcx> Visitor<'tcx> for InstrumentationAdder<'_, 'tcx> {
                     .debug_mir(location)
                     .add_to(self);
             }
-            Rvalue::Ref(_, bkind, p) if has_outer_deref(p) => {
+            Rvalue::Ref(_, bkind, p) => {
                 // this is a reborrow or field reference, i.e. _2 = &(*_1)
-                let source = remove_outer_deref(*p, self.tcx());
-                if let BorrowKind::Mut { .. } = bkind {
+                let source = try_remove_outer_deref(*p, self.tcx());
```

Comment:
i don't want to combine these, there are more complicated cases incoming and it'll be much clearer if they're separate